tests/unit/browser/test_history.py

```diff
@@ -229,6 +229,37 @@ def test_exclude(self, web_history, config_stub):
         assert list(web_history)
         assert not list(web_history.completion)
 
+    def test_no_immedate_duplicates(self, web_history, mock_time):
+        url = QUrl("http://example.com")
+        url2 = QUrl("http://example2.com")
+        web_history.add_from_tab(QUrl(url), QUrl(url), 'title')
+        hist = list(web_history)
+        assert list(web_history)
```

Comment:
Why do `list(web_history)` twice here? Couldn't this just be

```suggestion
        assert hist
```

instead?